logilica_weekly_report/download_pdfs.py

```diff
@@ -1,47 +1,57 @@
+    """Download a PDF file for each team's report using the Logilica Web UI"""
+    with sync_playwright() as playwright:
+        with playwright.chromium.launch(headless=True) as browser:
+            with browser.new_context(accept_downloads=True) as context:
+                page = context.new_page()
 
-    # Fill the login form using environment variables
-    page.get_by_role("button", name="Log in With Email").click()
-    page.locator("#domain").fill(os.getenv("LOGILICA_DOMAIN"))
-    page.locator("#email").fill(os.getenv("LOGILICA_EMAIL"))
-    page.locator("#password").fill(os.getenv("LOGILICA_PASSWORD"))
+                # Fill the login form using environment variables
+                logging.info("Logging into Logilica")
+                page.goto(LOGILICA_LOGIN)
+                page.get_by_role("button", name="Log in With Email").click()
+                page.locator("#domain").fill(os.getenv("LOGILICA_DOMAIN"))
+                page.locator("#email").fill(os.getenv("LOGILICA_EMAIL"))
+                page.locator("#password").fill(os.getenv("LOGILICA_PASSWORD"))
+                page.get_by_role("button", name="Login").click()
+                logging.debug("Login to Logilica complete")
 
-    page.get_by_role("button", name="Login").click()
-    page.get_by_role("link", name="Custom Reports").click()
+                page.get_by_role("link", name="Custom Reports").click()
+                export_pdfs(teams_config, page, download_path)
 
-    # Cycle though list of teams dashboards to download a pdf for each team
-    for team, dashboards in teams_data["teams"].items():
+
+def export_pdfs(
+    teams_config: dict[str, any], page: Page, download_path: pathlib.Path
+) -> None:
+    """Cycle though the list of team dashboards to download a PDF for each team"""
+    for team, dashboards in teams_config.items():
         for dashboard, options in dashboards["team_dashboards"].items():
+            logging.info("Downloading '%s' for %s", dashboard, team)
             page.get_by_role("link", name=dashboard).click()
             page.get_by_role("button", name="Export PDF").click()
             # Handle download
             with page.expect_download() as download_info:
-                page.get_by_role("button", name="Download").click()
-            download = download_info.value
-            download.save_as(download_path + options["Filename"])
-            # Close PDF view dialog
-            page.get_by_role("button", name="Close").click()
-
-    # Clean up
-    context.close()
-    browser.close()
-
-
-with sync_playwright() as playwright:
-    run(playwright)
+                page.get_by_role("button", name="Download").nth(1).click()
+                download = download_info.value
+                download.save_as(download_path / options["Filename"])
+                # Close PDF view dialog
+                page.get_by_role("button", name="Close").nth(1).click()
```

Comment:
@rnapoles-rh, I had to add the calls to `nth(1)` to lines 52 and 56, here.

Reply:
/lgtm